src/environment.rs

```diff
@@ -100,11 +102,9 @@ fn resolve_floki_root_and_config(
 }
 
 /// Resolve a directory for floki to use for user-global file (caches etc)
-fn get_floki_work_path() -> Result<path::PathBuf, Error> {
-    let root: path::PathBuf = env::var("HOME")
-        .unwrap_or(format!("/tmp/{}/", get_user_details()?.0))
-        .into();
-    Ok(root.join(".floki"))
+fn get_floki_work_path(uid: libc::uid_t) -> path::PathBuf {
+    let root: path::PathBuf = env::var("HOME").unwrap_or(format!("/tmp/{}/", uid)).into();
```

Comment:
Ugh, this /tmp/ here is annoying me. I don't want to patch it away to use std::env::temp_dir just yet because of the conflicts, but it's on my radar. Would love to see if floki Just Works when compiled on Windows...

Reply:
What conflicts did you have in mind? `std::env::temp_dir` does seem better. This folder is just used for volumes so changing this won't cause issues apart from potentially a bunch of rebuilds for users which don't have `HOME` in their environment.

You know my views on Windows :-). I've never been against patches to get it to work, within reason. I have no way to test it though, and its work I won't do, since I have no interest in running Windows.

I assume it more or less just works under WSL? From the normal command line thing on Windows I don't have high hopes, since there a bunch of POSIX calls dotted around. Maybe try it if you have patience to see what happens?

That aside, I was also looking at the `/tmp` thing and wondered why I thought that was a good idea. I don't necessarily think its a bad idea (if the permissions are correct), but maybe there is a better overall solution here. Also we should probably start by querying `XDG_CACHE_DIR` before `HOME` (causing volumes to change for more users in the process, though).

Reply:
We could also not use `/tmp` at all, and just error out.

Another possibility is providing a more controllable `FLOKI_VOLUME_CACHE` environment variable, so users can set it in their shells or something.

Reply:
I meant merge conflicts because we'd both be editing the same line :smile: I'm happy with just trying the simpler fix for now and maybe in future moving to something better.